mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java

```diff
@@ -84,11 +86,15 @@ public void delete(Mailbox mailbox) throws MailboxException {
 
     @Override
     public Mailbox findMailboxByPath(MailboxPath path) throws MailboxException {
-        ResultSet resultSet = session.execute(select(FIELDS).from(TABLE_NAME).where(eq(PATH, path.toString())));
-        if (resultSet.isExhausted()) {
-            throw new MailboxNotFoundException(path);
-        } else {
-            return mailbox(resultSet.one());
+        try {
+            ResultSet resultSet = session.execute(select(FIELDS).from(TABLE_NAME).where(eq(PATH, path.toString())));
+            if (resultSet.isExhausted()) {
+                throw new MailboxNotFoundException(path);
+            } else {
+                return mailbox(resultSet.one());
+            }
+        } catch (InvalidQueryException e) {
+            throw new TooLongMailboxNameException("too long mailbox name");
```

Comment:
That is the error we catch when over limit name. Do we need to more check its messge?
Because it had detailed message like this:
Index expression values may not be larger than 64K